src/platforms/common/usb_serial.c

```diff
@@ -195,8 +195,12 @@ void usb_serial_set_config(usbd_device *dev, uint16_t value)
 #endif
 
 	/* Serial interface */
-	usbd_ep_setup(
-		dev, CDCACM_UART_ENDPOINT, USB_ENDPOINT_ATTR_BULK, CDCACM_PACKET_SIZE / 2U, debug_serial_receive_callback);
+#if defined(USB_HS)
+	const uint16_t uart_epout_size = CDCACM_PACKET_SIZE;
```

Comment:
We strongly suspect this winds up wasting half the buffer on the F7, but as long as you think that's ok, we're happy with this.

Reply:
I assume this has to match the descriptors
https://github.com/blackmagic-debug/blackmagic/blob/c4c763e75ee09511621eb432618116e8ec2013c5/src/platforms/common/usb_descriptors.h#L200-L204
It does not waste half the Tx FIFO, that is, I think I saw usbmon packets for 511 bytes in both directions; otherwise I'd see 255 bytes in one of directions.
You're right in that I should be more careful with FIFO memory allocation to non-periodic buffers, this isn't Linux afterall.

Reply:
Correct, it has to match the descriptors. As long as this patch does indeed make them match (which we just checked, it does), that's fine with us. Have to admit we didn't check, which is part of the query. We'll consider this resolved.